File: golang/algorithms/fibonacci.go
package algorithms

import "math/big"

func FibRec(n int) int {
	if n <= 1 {
		return n
	} else {
		return FibRec(n-1) + FibRec(n-2)
	}
}

func FibIter(n int) []int {
	row := []int{0, 1}
	for i := 0; i < n-2; i++ {
		value := row[len(row)-1] + row[len(row)-2]
		row = append(row, value)
	}

	return row
}

func FibIter2(n uint) *big.Int {
	a, b := big.NewInt(0), big.NewInt(1)
	for i := uint(1); i < n; i++ {
		a.Add(a, b)
		a, b = b, a
	}

	return b
}

File: python/algorithms/fibonacci.py
# Ряд фибоначчи


def fib(number):
    """
    Генератор
    :param number:
    :return:
    """
    a, b = 0, 1

    for _ in range(number):
        yield a
        a, b = b, a + b


def fib_iter(number):
    """
    Итеративный ряд фибоначчи
    :param number:
    :return:
    """
    row = [0, 1]
    for n in range(number - 2):
        row.append(row[-1] + row[-2])

    return row


def fib_rec(number):
    """
    Рекурсивное вычисление числа фибоначчи
    :param number:
    :return:
    """
    if number <= 1:
        return number
    else:
        return fib_rec(number - 1) + fib_rec(number - 2)


if __name__ == '__main__':
    print(fib_rec(10))
    print(fib_iter(10))
    print(list(fib(1200)))
